### PR TITLE
[15.0][MIG] mail_notification_custom_subject: Migration to 15.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 15.0

Please @CarlosRoca13 and @chienandalu can you review it?

@Tecnativa TT36470

### CODE REVIEW

**Review by CarlosRoca13 — APPROVED** [review]

File: mail_notification_custom_subject/models/mail_thread.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        parent_id=False,
+        subtype_xmlid=None,
+        subtype_id=False,
+        partner_ids=None,
+        attachments=None,
+        attachment_ids=None,
+        add_sign=True,
+        record_name=False,
+        **kwargs
+    ):
```
Comment (chienandalu) [inline mail_notification_custom_subject/models/mail_thread.py:29]: Simpler:
```suggestion
    def message_post(self, **kwargs):
```

File: mail_notification_custom_subject/models/mail_thread.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            parent_id=parent_id,
+            subtype_xmlid=subtype_xmlid,
+            subtype_id=subtype_id,
+            partner_ids=partner_ids,
+            attachments=attachments,
+            attachment_ids=attachment_ids,
+            add_sign=add_sign,
+            record_name=record_name,
+            **kwargs,
+        )
```
Comment (chienandalu) [inline mail_notification_custom_subject/models/mail_thread.py:78]:
```suggestion
        return super().message_post(**kwargs)
```

File: mail_notification_custom_subject/models/mail_thread.py [hunk 3]
```diff
@@ -21,15 +21,14 @@ def message_post(
...
         subtype_id=False,
         partner_ids=None,
-        channel_ids=None,
         attachments=None,
         attachment_ids=None,
         add_sign=True,
         record_name=False,
         **kwargs
     ):
         if not subtype_id and subtype_xmlid:
```
Comment (chienandalu) [inline mail_notification_custom_subject/models/mail_thread.py:30]:
```suggestion
        subtype_id = kwargs.pop("subtype_id", False)
        subtype_xmlid = kwargs.pop("subtype_xmlid", False)
        if not subtype_id and subtype_xmlid:
```

**Review by chienandalu — COMMENTED** [review]
Tested functionaly. Some formal comments:

**Comment by victoralmau** [comment]
> Tested functionaly. Some formal comments:

Changes done, but it seems to have side effects on other modules. Any suggestions to solve it?

**Comment by pedrobaeza** [comment]
You can't do that, as not the whole inheritance chain is respecting keyword arguments, so better to stick to the whole form.

**Comment by victoralmau** [comment]
Changes done (last changes undone)

**Comment by chienandalu** [comment]
> You can't do that, as not the whole inheritance chain is respecting keyword arguments, so better to stick to the whole form.

Odoo's choice, not mine :smile: :

- https://github.com/odoo/odoo/blob/15.0/addons/rating/models/mail_thread.py#L11
- https://github.com/odoo/odoo/blob/15.0/addons/hr/models/hr_employee.py#L457
- https://github.com/odoo/odoo/blob/15.0/addons/purchase/models/purchase.py#L412
- ...

**Comment by pedrobaeza** [comment]
Well, that ones probably will raise an error with this same chain... Maybe it's a problem in one OCA module, but that form requires that you always call super with all the keywords.

**Pull request merged**